runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/NoHttpEngine.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.engine
+
+import aws.smithy.kotlin.runtime.http.config.EngineFactory
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.response.HttpCall
+import aws.smithy.kotlin.runtime.operation.ExecutionContext
+
+/**
+ * Provides an implementation of an [HttpClientEngine] which is guaranteed never to make actual network calls. (Attempts
+ * to make network calls result in an exception.) Use this when you need to configure a service client, but you are
+ * certain no HTTP engine is actually needed.
+ */
+public class NoHttpEngine : HttpClientEngineBase("none") {
```

Comment:
**Question**: I know we discussed this as an option but I didn't necessarily think we would own it/provide it. I imagined it more as an escape hatch available if absolutely needed. Is this something we actually need to provide out of the box?

Reply:
Ah then I misunderstood. I found it useful to have a `NoHttpEngine` implementation when updating the [S3](https://github.com/awslabs/aws-sdk-kotlin/pull/921/files#diff-4d9bd50b3adffc65d327527a09402127b64f92c1a9d13ad84e894ba939f0e308) and [Polly](https://github.com/awslabs/aws-sdk-kotlin/pull/921/files#diff-dd429df13fe7c3ec049d98dbbf70c8f86c3266e904b01c6362f93ab252e050d0) presigning unit tests but I could always make this a test-only utility.

Reply:
I see, I'm not 100% opposed I just hadn't pictured us shipping it. I don't necessarily see harm in it. We could discuss

Reply:
Discussed offline and decided to _not_ publish `NoHttpEngine` for now. We can revisit when we get more specific user requests.